src/MongoDB.Bson/IO/JsonReader.cs

```diff
@@ -1330,7 +1330,9 @@ private BsonValue ParseDateTimeConstructor(bool withNew)
 
         private BsonType ParseExtendedJson()
         {
+            var bookmark = GetBookmark();
```

Comment:
I had the same thought when reviewing, but... in the future it could easily be required for other extended JSON formats so I didn't object to it being done this generally.

It is perfectly safe and bookmarks are cheap so no performance concerns either.

Reply:
Got it